test/Constraints/optional.swift

```diff
@@ -565,3 +565,26 @@ func testFunctionContainerMethodCall(container: FunctionContainer?) {
   // expected-note@-3 {{coalesce}}
   // expected-note@-4 {{force-unwrap}}
 }
+
+// Test for https://github.com/apple/swift/issues/60730
+// rdar://94037733
+do {
+  struct S: P {}
+  func takesP(_: any P) {}
+  func passOptional(value: (any P)?) {
+    takesP(value)
+    // expected-error@-1 {{value of optional type '(any P)?' must be unwrapped to a value of type 'any P'}}
+    // expected-note@-2 {{coalesce using '??' to provide a default when the optional value contains 'nil'}}
+    // expected-note@-3 {{force-unwrap using '!' to abort execution if the optional value contains 'nil'}}
+  }
+  func passLayersOfOptional(value: (any P)??) {
+    takesP(value)
+    // expected-error@-1 {{value of optional type '(any P)??' must be unwrapped to a value of type '(any P)?}}
```

Comment:
Hm, this is not great... Could you please leave a `FIXME(diagnostics):` here to to record multiple ForceUnwrap fixes based on number of optionals we need to unwrap?

Reply:
Where do I add the "FIXME"? As a comment above the takesP() call?

Reply:
Yes, that's good enough.